### PR TITLE
ADCM-3267 Rework `adcm` endpoint

### PR DESCRIPTION
**Pull request by a-alferov**

### CODE REVIEW

File: python/api/adcm/urls.py [hunk 1]
```diff
@@ -12,18 +12,11 @@
 
 from django.urls import include, path
 
-from api.adcm.views import AdcmDetail, AdcmList
+from .views import ADCMViewSet
```
Comment (winmasta) [inline python/api/adcm/urls.py:15]: Explicit import is better

File: python/api/adcm/urls.py [hunk 2]
```diff
@@ -12,18 +12,11 @@
...
-        "<int:adcm_id>/",
-        include(
-            [
-                path("", AdcmDetail.as_view(), name="adcm-details"),
-                path("config/", include("api.config.urls"), {"object_type": "adcm"}),
-                path("action/", include("api.action.urls"), {"object_type": "adcm"}),
-            ]
-        ),
-    ),
+    path(r"", ADCMViewSet.as_view({"get": "list"}), name="adcm-list"),
```
Comment (winmasta) [inline python/api/adcm/urls.py:18]: `r` is redundant probably

File: python/api/action/views.py [hunk 3]
```diff
@@ -48,7 +48,8 @@
 
 def get_object_type_id(**kwargs):
     object_type = kwargs.get('object_type')
-    object_id = kwargs.get(f'{object_type}_id')
+    # TODO: this is a temporary patch for `action` endpoint
+    object_id = kwargs.get(f'{object_type}_id') or kwargs.get(f"{object_type}_pk")
```
Comment (winmasta) [inline python/api/action/views.py:52]: Double quotas can be used

File: python/api/adcm/views.py [hunk 4]
```diff
@@ -10,45 +10,33 @@
...
-    serializer_class = AdcmDetailSerializer
-    serializer_class_ui = AdcmDetailUISerializer
-    permission_classes = (permissions.IsAuthenticated,)
-    lookup_field = 'id'
-    lookup_url_kwarg = 'adcm_id'
-    error_code = 'ADCM_NOT_FOUND'
+# pylint:disable-next=too-many-ancestors
+class ADCMViewSet(ListModelMixin, RetrieveModelMixin, GenericUIViewSet):
+
+    # queryset = ADCM.objects.all()
```
Comment (winmasta) [inline python/api/adcm/views.py:28]: Commented code

File: python/api/base_view.py [hunk 5]
```diff
@@ -79,8 +79,15 @@ def get_serializer_class(self):
...
-class GenericUIViewSet(ViewSetMixin, GenericUIView):
-    """GenericUIView with expanded for ViewSet"""
+class GenericUIViewSet(ViewSetMixin, GenericAPIView):
+
+    permission_classes = (DjangoObjectPermissionsAudit,)
+
+    def is_for_ui(self) -> bool:
+        if not self.request:
+            return False
+        view = self.request.query_params.get("view", None)
```
Comment (winmasta) [inline python/api/base_view.py:89]: `get` returns None by default as I remember

File: python/api/group_config/views.py [hunk 6]
```diff
@@ -204,7 +208,7 @@ def get_serializer_context(self):
             obj_ref = ObjectConfig.obj.get(id=obj_ref_id)
             context.update({'obj_ref': obj_ref})
 
-        context['ui'] = self._is_for_ui()
+        context['ui'] = self.is_for_ui()
```
Comment (winmasta) [inline python/api/group_config/views.py:211]: Double quotas van be used

File: python/api/adcm/urls.py (hunk 2 above)
Comment (a-alferov) [reply python/api/adcm/urls.py]: fix

File: python/api/action/views.py (hunk 3 above)
Comment (a-alferov) [reply python/api/action/views.py]: fix

File: python/api/adcm/views.py (hunk 4 above)
Comment (a-alferov) [reply python/api/adcm/views.py]: fixed

File: python/api/base_view.py (hunk 5 above)
Comment (a-alferov) [reply python/api/base_view.py]: fixed

File: python/api/group_config/views.py (hunk 6 above)
Comment (a-alferov) [reply python/api/group_config/views.py]: fixed

**Pull request closed without merging**